srcpkgs/gimx/template

```diff
@@ -0,0 +1,65 @@
+# Template file for 'gimx'
+pkgname=gimx
+version=8.0
+revision=1
+build_style=gnu-makefile
+make_use_env=yes
+hostmakedepends="gettext pkg-config"
+makedepends="libbluetooth-devel libcurl-devel libusb-devel libX11-devel libXi-devel
+ libxml2-devel mhash-devel ncurses-devel wxWidgets-gtk3-devel"
+depends="gimx-cli-${version}_${revision} gimx-firmware-${version}_${revision}
+ avrdude xdg-utils xterm"
+short_desc="Use a computer as a hub for your gaming devices"
+maintainer="amak <amak.git@outlook.com>"
+license="GPL-3.0-only"
+homepage="https://blog.gimx.fr/"
+distfiles="https://github.com/matlo/GIMX/releases/download/v${version}/gimx_${version}-1.tar.gz"
+checksum=783673435bd424ba49e6ad5911559a517f746e50dad0f10689f0dc53336a8e1e
+patch_args="-Np1"
+
+# silence unused function errors in loader/gimx-loader.cpp
+# that are triggered by archs not defined in info.h
+CXXFLAGS=" -Wno-unused-function"
+
+if [ "$CROSS_BUILD" ]; then
+	hostmakedepends+=" ncurses-devel"
+fi
+
+post_extract() {
+	# fix build with ncurses 6
+	vsed -i -e "s:ncursesw5-config:ncursesw6-config:" core/Makefile
+
+	# fix ncursesw/ncurses.h include
+	vsed -i -e "s:<ncursesw/ncurses.h>:<curses.h>:" core/display.c
+
+	# fix build with wxWidgets-gtk3
+	find . -iname Makefile -exec sed -i -e "s:wx-config:&-gtk3:" "{}" \;
+}
+
+post_install() {
+	vdoc README.md
+}
+
+gimx-cli_package() {
+	short_desc+=" - cli tools"
+	shlib_provides="libgimxconfigeditor.so libgimxconfigupdater.so libgimxcontroller.so
+	 libgimxdownloader.so libgimxfile.so libgimxgpp.so libgimxhid.so libgimxinput.so
+	 libgimxlog.so libgimxpoll.so libgimxprio.so libgimxserial.so libgimxtimer.so
+	 libgimxtime.so libgimxudp.so libgimxuhid.so libgimxupdater.so libgimxusb.so"
```

Comment:
There's something wonky making this necessary, because it really shouldn't be. I'd like to look into it before merging.

Reply:
With `shlib_provides` removed I get the following when installing the `gimx` package.
```
# xbps-install gimx
gimx-8.0_1: broken, unresolvable shlib `libgimxconfigeditor.so'
gimx-8.0_1: broken, unresolvable shlib `libgimxconfigupdater.so'
gimx-8.0_1: broken, unresolvable shlib `libgimxcontroller.so'
gimx-8.0_1: broken, unresolvable shlib `libgimxfile.so'
gimx-8.0_1: broken, unresolvable shlib `libgimxgpp.so'
gimx-8.0_1: broken, unresolvable shlib `libgimxhid.so'
gimx-8.0_1: broken, unresolvable shlib `libgimxinput.so'
gimx-8.0_1: broken, unresolvable shlib `libgimxpoll.so'
gimx-8.0_1: broken, unresolvable shlib `libgimxserial.so'
gimx-8.0_1: broken, unresolvable shlib `libgimxtime.so'
gimx-8.0_1: broken, unresolvable shlib `libgimxupdater.so'
Transaction aborted due to unresolved shlibs.
```
I don't get this error when installing the `gimx-cli` package. Could it be caused by the way I have split the package? I move all the shared libraries into the `gimx-cli` package, even those that aren't needed by `gimx-cli`. I figured that was ok since `gimx` has a depend on `gimx-cli`.

Reply:
Many ignored runtime deps for `gimx-cli`.
```
=> gimx-cli-8.0_1: running pre-pkg hook: 04-generate-runtime-deps ...
   SONAME: libstdc++.so.6 <-> libstdc++>=4.4.0_1
   SONAME: libcurl.so.4 <-> libcurl>=7.75.0_2
   SONAME: libgimxfile.so <-> gimx-cli-8.0_1 (ignored)
   SONAME: libgcc_s.so.1 <-> libgcc>=4.4.0_1
   SONAME: libc.so.6 <-> glibc>=2.32_1
   SONAME: libgimxlog.so <-> gimx-cli-8.0_1 (ignored)
   SONAME: libgimxdownloader.so <-> gimx-cli-8.0_1 (ignored)
   SONAME: libgimxhid.so <-> gimx-cli-8.0_1 (ignored)
   SONAME: libusb-1.0.so.0 <-> libusb>=1.0.0_1
   SONAME: libxml2.so.2 <-> libxml2>=2.7.0_1
   SONAME: libm.so.6 <-> glibc>=2.32_1
   SONAME: libgimxinput.so <-> gimx-cli-8.0_1 (ignored)
   SONAME: libgimxtimer.so <-> gimx-cli-8.0_1 (ignored)
   SONAME: libgimxpoll.so <-> gimx-cli-8.0_1 (ignored)
   SONAME: libgimxcontroller.so <-> gimx-cli-8.0_1 (ignored)
   SONAME: libgimxuhid.so <-> gimx-cli-8.0_1 (ignored)
   SONAME: libXi.so.6 <-> libXi>=1.2.1_1
   SONAME: libX11.so.6 <-> libX11>=1.2_1
   SONAME: libgimxusb.so <-> gimx-cli-8.0_1 (ignored)
   SONAME: libbluetooth.so.3 <-> libbluetooth>=4.58_1
   SONAME: libgimxgpp.so <-> gimx-cli-8.0_1 (ignored)
   SONAME: libgimxtime.so <-> gimx-cli-8.0_1 (ignored)
   SONAME: libgimxprio.so <-> gimx-cli-8.0_1 (ignored)
   SONAME: libgimxserial.so <-> gimx-cli-8.0_1 (ignored)
   SONAME: libgimxudp.so <-> gimx-cli-8.0_1 (ignored)
   SONAME: libncursesw.so.6 <-> ncurses-libs>=5.8_1
   SONAME: libmhash.so.2 <-> mhash>=0.9.9.9_1
```